### DISCUSSION TITLE
Issue with domain pointing for frontend

### DISCUSSION THREAD

**Discussion by Neoblock8** [blockscout/blockscout#8921] [General]
Hi,
 Running blockscout backend with 4000 port and point with domain ['https://example.com'].
 And I ran blockscout frontend with 3000 port. Its working perfectly with http://<ip>:3000.
 But configure with **"https://example.com"** the URL, frontend is not working. Showing data is not found

**Comment by ekawibisono** [comment]
Hello, I see this message the same as on discord, please choose whether to resolve via github discussion or on discord?

**Comment by Neoblock8** [comment]
> Hello, I see this message the same as on discord, please choose whether to resolve via github discussion or on discord?

Kindly provide solution in github

**Reply by ekawibisono** [reply]
Please provide your detail problem with screenshot image or video

**Comment by Neoblock8** [comment]
Hi @ekawibisono,

If nginx configure with 4000 port with domain pointing [https://example.com]

IPs:
https://example.com - working good
http://<ip>:4000 - working good
http://<ip>:3000 - working good

If nginx configure with 3000 port and i configure domain in .env file under frontend folder
 IPs:
https://example.com - working good
http://<ip>:4000 - working good
http://<ip>:3000 - not working. 

Error - Data not found

This is my env file:
# app config
NEXT_PUBLIC_APP_PROTOCOL=https
NEXT_PUBLIC_APP_HOST=example.com
NEXT_PUBLIC_APP_PORT=443
NEXT_PUBLIC_APP_INSTANCE=local
NEXT_PUBLIC_APP_ENV=development
NEXT_PUBLIC_SENTRY_DSN=
#NEXT_PUBLIC_IS_TESTNET=true

# ui config
#NEXT_PUBLIC_FEATURED_NETWORKS=PLACEHOLDER_FOR_NEXT_PUBLIC_FEATURED_NETWORKS
#NEXT_PUBLIC_NETWORK_EXPLORERS=PLACEHOLDER_FOR_NEXT_PUBLIC_NETWORK_EXPLORERS
NEXT_PUBLIC_HOMEPAGE_CHARTS=['daily_txs','coin_price','market_cap']
#NEXT_PUBLIC_API_SPEC_URL=PLACEHOLDER_FOR_NEXT_PUBLIC_NETWORK_EXPLORERS
#NEXT_PUBLIC_MARKETPLACE_CONFIG_URL=
#NEXT_PUBLIC_MARKETPLACE_SUBMIT_FORM=
NEXT_PUBLIC_HOMEPAGE_PLATE_TEXT_COLOR=\#DCFE76
NEXT_PUBLIC_HOMEPAGE_PLATE_BACKGROUND=radial-gradient(103.03% 103.03% at 0% 0%, rgba(183, 148, 244, 0.8) 0%, rgba(0, 163, 196, 0.8) 100%), var(--chakra-colors-blue-400)
NEXT_PUBLIC_AD_BANNER_PROVIDER=none

# network config
NEXT_PUBLIC_NETWORK_NAME=TEST
NEXT_PUBLIC_NETWORK_SHORT_NAME=TST
NEXT_PUBLIC_NETWORK_ID=123
NEXT_PUBLIC_NETWORK_CURRENCY_NAME=TST
NEXT_PUBLIC_NETWORK_CURRENCY_SYMBOL=TST
NEXT_PUBLIC_NETWORK_CURRENCY_DECIMALS=18
NEXT_PUBLIC_NETWORK_VERIFICATION_TYPE=validation
NEXT_PUBLIC_NETWORK_RPC_URL=http://127.0.0.1:8545
#NEXT_PUBLIC_IS_ACCOUNT_SUPPORTED=true
#NEXT_PUBLIC_AUTH0_CLIENT_ID=
#NEXT_PUBLIC_AUTH_URL=https://example.com/
#NEXT_PUBLIC_LOGOUT_URL=https://example.com/api/v2/logout

# api config
NEXT_PUBLIC_API_BASE_PATH=/
NEXT_PUBLIC_API_HOST=example.com
NEXT_PUBLIC_API_PROTOCOL=https
NEXT_PUBLIC_API_PORT=443
NEXT_PUBLIC_API_WEBSOCKET_PROTOCOL=wss

**Reply by ekawibisono** [reply]
@Neoblock8 why port SSL in here ? `NEXT_PUBLIC_API_PORT=443`

**Reply by Neoblock8** [reply]
Then, what i have to give?

**Comment by ekawibisono** [comment]
Please provide your file default.conf.template on folder proxy @Neoblock8

**Comment by Neoblock8** [comment]
> @Neoblock8 why port SSL in here ? `NEXT_PUBLIC_API_PORT=443`



> default.conf.template


map $http_upgrade $connection_upgrade {

  default upgrade;
  ''      close;
}

server {
    listen       80;
    server_name  localhost;
    proxy_http_version 1.1;

    location / {
        proxy_pass            ${BACK_PROXY_PASS};
        proxy_http_version    1.1;
        proxy_set_header      Host "$host";
        proxy_set_header      X-Real-IP "$remote_addr";
        proxy_set_header      X-Forwarded-For "$proxy_add_x_forwarded_for";
        proxy_set_header      X-Forwarded-Proto "$scheme";
        proxy_set_header      Upgrade "$http_upgrade";
        proxy_set_header      Connection $connection_upgrade;
        proxy_cache_bypass    $http_upgrade;
    }
    location = / {
        proxy_pass            ${FRONT_PROXY_PASS};
        proxy_http_version    1.1;
        proxy_set_header      Host "$host";
        proxy_set_header      X-Real-IP "$remote_addr";
        proxy_set_header      X-Forwarded-For "$proxy_add_x_forwarded_for";
        proxy_set_header      X-Forwarded-Proto "$scheme";
        proxy_set_header      Upgrade "$http_upgrade";
        proxy_set_header      Connection $connection_upgrade;
        proxy_cache_bypass    $http_upgrade;
    }
    location ~ ^/(_next|node-api|apps|account|accounts|static|auth/profile|auth/unverified-email|txs|tx|blocks|block|login|address|stats|search-results|token|tokens|visualize|api-docs|csv-export|verified-contracts|graphiql|withdrawals) {
        proxy_pass            ${FRONT_PROXY_PASS};
        proxy_http_version    1.1;
        proxy_set_header      Host "$host";
        proxy_set_header      X-Real-IP "$remote_addr";
        proxy_set_header      X-Forwarded-For "$proxy_add_x_forwarded_for";
        proxy_set_header      X-Forwarded-Proto "$scheme";
        proxy_set_header      Upgrade "$http_upgrade";
        proxy_set_header      Connection $connection_upgrade;
        proxy_cache_bypass    $http_upgrade;
    }
}
server {
    listen       8080;
    server_name  localhost;
    proxy_http_version 1.1;
    proxy_hide_header Access-Control-Allow-Origin;
    proxy_hide_header Access-Control-Allow-Methods;
    add_header 'Access-Control-Allow-Origin' 'http://localhost' always;
    add_header 'Access-Control-Allow-Credentials' 'true' always;
    add_header 'Access-Control-Allow-Methods' 'PUT, GET, POST, OPTIONS, DELETE, PATCH' always;

    location / {
        proxy_pass            http://stats:8050;
        proxy_http_version    1.1;
        proxy_set_header      Host "$host";
        proxy_set_header      X-Real-IP "$remote_addr";
        proxy_set_header      X-Forwarded-For "$proxy_add_x_forwarded_for";
        proxy_set_header      X-Forwarded-Proto "$scheme";
        proxy_set_header      Upgrade "$http_upgrade";
        proxy_set_header      Connection $connection_upgrade;
        proxy_cache_bypass    $http_upgrade;
    }
}

server {
    listen       8090;
    server_name  localhost;
    proxy_http_version 1.1;
    proxy_hide_header Access-Control-Allow-Origin;
    proxy_hide_header Access-Control-Allow-Methods;
    proxy_hide_header Access-Control-Allow-Headers;
    add_header 'Access-Control-Allow-Origin' 'http://localhost' always;
    add_header 'Access-Control-Allow-Credentials' 'true' always;
    add_header 'Access-Control-Allow-Methods' 'PUT, GET, POST, OPTIONS, DELETE, PATCH' always;
    add_header 'Access-Control-Allow-Headers' 'DNT,Keep-Alive,User-Agent,X-Requested-With,If-Modified-Since,Cache-Control,Content-Type,Range,Authorization,x-csrf-token' always;

    location / {
        proxy_pass            http://visualizer:8050;
        proxy_http_version    1.1;
        proxy_set_header      Host "$host";
        proxy_set_header      X-Real-IP "$remote_addr";
        proxy_set_header      X-Forwarded-For "$proxy_add_x_forwarded_for";
        proxy_set_header      X-Forwarded-Proto "$scheme";
        proxy_set_header      Upgrade "$http_upgrade";
        proxy_set_header      Connection $connection_upgrade;
        proxy_cache_bypass    $http_upgrade;
    }
}

**Comment by ekawibisono** [comment]
@Neoblock8 you use blockscout version ?

**Comment by ekawibisono** [comment]
@Neoblock8 i see you using blockscout old, please use blockscout new version V5.3.2

**Comment by Neoblock8** [comment]
In new blockscout, can we run both frontend and backend

**Comment by ekawibisono** [comment]
@Neoblock8 yes, you can run backend and frontend

**Comment by Neoblock8** [comment]
I configure everything in this version. Please provide a solution

**Comment by ekawibisono** [comment]
@Neoblock8 I think you have a problem with your proxy. 
Maybe I can help you by attaching a link to a video I made from google drive that uses the old blockscout version. maybe you just watch and learn. and you can implement it in the latest version. [Video Tutorials](https://drive.google.com/drive/folders/1dH-ZxV1ShQ-cb7gObTXs3MtwSqVNJOqW?usp=drive_link)
There is a choice of 2 blockscout installation methods, please choose the one you use, between using docker compose or manual build.

**Reply by Neoblock8** [reply]
Both backend and frontend is working good with URL [http://:4000 & http://:3000].
I got a problem while connecting with nginx

**Reply by ekawibisono** [reply]
i send my configure nginx proxy for you

**Comment by ekawibisono** [comment]
this my config nginx proxy @Neoblock8 you can implement it yourself
```
map $http_upgrade $connection_upgrade {

  default upgrade;
  ''      close;
}

server {
    listen       9767;
    listen       443 ssl;
    server_name  frontend.testingtoken.tech;
    proxy_http_version 1.1;

    ssl_certificate /etc/frontend/frontend.pem;
    ssl_certificate_key /etc/frontend/frontend.key;

    location ~ ^/(api|socket|sitemap.xml|auth/auth0|auth/auth0/callback|auth/logout) {
        proxy_pass            ${BACK_PROXY_PASS};
        proxy_http_version    1.1;
        proxy_set_header      Host "$host";
        proxy_set_header      X-Real-IP "$remote_addr";
        proxy_set_header      X-Forwarded-For "$proxy_add_x_forwarded_for";
        proxy_set_header      X-Forwarded-Proto "$scheme";
        proxy_set_header      Upgrade "$http_upgrade";
        proxy_set_header      Connection $connection_upgrade;
        proxy_cache_bypass    $http_upgrade;
    }
    location / {
        proxy_pass            http://128.199.129.190:3000;
        proxy_http_version    1.1;
        proxy_set_header      Host "$host";
        proxy_set_header      X-Real-IP "$remote_addr";
        proxy_set_header      X-Forwarded-For "$proxy_add_x_forwarded_for";
        proxy_set_header      X-Forwarded-Proto "$scheme";
        proxy_set_header      Upgrade "$http_upgrade";
        proxy_set_header      Connection $connection_upgrade;
        proxy_cache_bypass    $http_upgrade;
    }
}
server {
    listen       8080;
    listen       443 ssl;
    server_name  stats.testingtoken.tech;
    proxy_http_version 1.1;

    ssl_certificate /etc/stats/stats.pem;
    ssl_certificate_key /etc/stats/stats.key;

    proxy_hide_header Access-Control-Allow-Origin;
    proxy_hide_header Access-Control-Allow-Methods;
    add_header 'Access-Control-Allow-Origin' '*' always;
    add_header 'Access-Control-Allow-Credentials' 'true' always;
    add_header 'Access-Control-Allow-Methods' 'PUT, GET, POST, OPTIONS, DELETE, PATCH' always;

    location / {
        proxy_pass            http://stats:8050/;
        proxy_http_version    1.1;
        proxy_set_header      Host "$host";
        proxy_set_header      X-Real-IP "$remote_addr";
        proxy_set_header      X-Forwarded-For "$proxy_add_x_forwarded_for";
        proxy_set_header      X-Forwarded-Proto "$scheme";
        proxy_set_header      Upgrade "$http_upgrade";
        proxy_set_header      Connection $connection_upgrade;
        proxy_cache_bypass    $http_upgrade;
    }
}
server {
    listen       8081;
    listen       443 ssl;
    server_name  visualize.testingtoken.tech;

    ssl_certificate /etc/visualize/visualize.pem;
    ssl_certificate_key /etc/visualize/visualize.key;

    proxy_http_version 1.1;
    proxy_hide_header Access-Control-Allow-Origin;
    proxy_hide_header Access-Control-Allow-Methods;
    add_header 'Access-Control-Allow-Origin' '*' always;
    add_header 'Access-Control-Allow-Credentials' 'true' always;
    add_header 'Access-Control-Allow-Methods' 'PUT, GET, POST, OPTIONS, DELETE, PATCH' always;
    add_header 'Access-Control-Allow-Headers' 'DNT,Keep-Alive,User-Agent,X-Requested-With,If-Modified-Since,Cache-Control,Content-Type,Range,Authorization,x-csrf-token' always;

    location / {
        proxy_pass            http://visualizer:8050/;
        proxy_http_version    1.1;
        proxy_buffering       off;
        proxy_set_header      Host "$host";
        proxy_set_header      X-Real-IP "$remote_addr";
        proxy_connect_timeout 30m;
        proxy_read_timeout    30m;
        proxy_send_timeout    30m;
        proxy_set_header      X-Forwarded-For "$proxy_add_x_forwarded_for";
        proxy_set_header      X-Forwarded-Proto "$scheme";
        proxy_set_header      Upgrade "$http_upgrade";
        proxy_set_header      Connection $connection_upgrade;
        proxy_cache_bypass    $http_upgrade;
        if ($request_method = 'OPTIONS') {
            add_header 'Access-Control-Allow-Origin' '*' always;
            add_header 'Access-Control-Allow-Credentials' 'true' always;
            add_header 'Access-Control-Allow-Methods' 'PUT, GET, POST, OPTIONS, DELETE, PATCH' always;
            add_header 'Access-Control-Allow-Headers' 'DNT,Keep-Alive,User-Agent,X-Requested-With,If-Modified-Since,Cache-Control,Content-Type,Range,Authorization,x-csrf-token' always;
            add_header 'Access-Control-Max-Age' 1728000;
            add_header 'Content-Type' 'text/plain charset=UTF-8';
            add_header 'Content-Length' 0;
            return 204;
        }
    }
}

server {
    listen       9709;
    listen       443 ssl;
    server_name  rpc.testingtoken.tech;
    proxy_http_version 1.1;

    ssl_certificate /etc/rpc/rpc.pem;
    ssl_certificate_key /etc/rpc/rpc.key;

    location / {
        proxy_pass            http://89.117.52.155:9545/;
        proxy_http_version    1.1;
        proxy_set_header      Host "$host";
        proxy_set_header      X-Real-IP "$remote_addr";
        proxy_set_header      X-Forwarded-For "$proxy_add_x_forwarded_for";
        proxy_set_header      X-Forwarded-Proto "$scheme";
        proxy_set_header      Upgrade "$http_upgrade";
        proxy_set_header      Connection $connection_upgrade;
        proxy_cache_bypass    $http_upgrade;
    }
}
```

**Comment by Neoblock8** [comment]
What did you set in .env file for NEXT_PUBLIC_API_PORT=443

**Reply by ekawibisono** [reply]
API PORT you check on your proxy, if you using 80, not add NEXT_PUBLIC_API_PORT

**Reply by Sohailghafoorj** [reply]
Can you tell me for production only default.conf file need to edit and use or also need to use microservices.conf file
@Neoblock8 @ekawibisono @atoulme @rkachowski @konstantinzolotarev

**Reply by ekawibisono** [reply]
hello @Sohailghafoorj please visit discord blockscout [Blockscout Discord](https://discord.gg/blockscout)
